### PR TITLE
texlive: fix dependency bug, add docs and release notes

### PR DESCRIPTION
**Pull request by xworld21**
## Description of changes
This contains an urgent fix for an embarrassing dependency resolution bug when using old style custom packages.

It also contains updated documentation about custom TeX packages (which is how I discovered the bug) and a line in the 23.11 release notes about `texlive.withPackages`.

If we miss the branch off, let's remember to backport to 23.11.

## Things done
- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg build tests.texlive

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 1]
```diff
@@ -551,6 +551,8 @@ The module update takes care of the new config syntax and the data itself (user
 
 - The argument `vendorSha256` of `buildGoModule` is deprecated. Use `vendorHash` instead. ([\#259999](https://github.com/NixOS/nixpkgs/pull/259999))
 
+- TeX Live environments can now be built with the new `texlive.withPackages` and creating custom TeX packages has been simplified.
```
Comment (veprbl) [inline nixos/doc/manual/release-notes/rl-2311.section.md:554]:
```suggestion
- TeX Live environments can now be built with the new `texlive.withPackages`. Procedure for creating custom TeX packages has been changed, see the [Nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/#sec-language-texlive-custom-packages) for more details.
```

Comment (xworld21) [reply nixos/doc/manual/release-notes/rl-2311.section.md:554]: Should we also say `Existing custom packages are still supported.`?

**Review by veprbl — APPROVED** [review]

File: nixos/doc/manual/release-notes/rl-2311.section.md (hunk 1 above)
Comment (veprbl) [reply nixos/doc/manual/release-notes/rl-2311.section.md]: It's probably not. There are no tests, and I assume, you plan to deprecate/remove it.

**Pull request merged**